taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -1492,20 +1560,49 @@ llvm::Value *CodeGenLLVM::create_bit_ptr(llvm::Value *byte_ptr,
   // 3. store `byte_ptr`
   builder->CreateStore(
       byte_ptr, builder->CreateGEP(
+#ifdef TI_LLVM_15
+                    struct_type,
+#endif
                     bit_ptr, {tlctx->get_constant(0), tlctx->get_constant(0)}));
   // 4. store `bit_offset
-  builder->CreateStore(bit_offset,
-                       builder->CreateGEP(bit_ptr, {tlctx->get_constant(0),
-                                                    tlctx->get_constant(1)}));
+  builder->CreateStore(
+      bit_offset,
+      builder->CreateGEP(
+#ifdef TI_LLVM_15
+          struct_type,
+#endif
+          bit_ptr, {tlctx->get_constant(0), tlctx->get_constant(1)}));
   return bit_ptr;
 }
 
 std::tuple<llvm::Value *, llvm::Value *> CodeGenLLVM::load_bit_ptr(
     llvm::Value *bit_ptr) {
-  auto byte_ptr = builder->CreateLoad(builder->CreateGEP(
-      bit_ptr, {tlctx->get_constant(0), tlctx->get_constant(0)}));
-  auto bit_offset = builder->CreateLoad(builder->CreateGEP(
-      bit_ptr, {tlctx->get_constant(0), tlctx->get_constant(1)}));
+#ifdef TI_LLVM_15
```

Comment:
Looks like there are too many TI_LLVM_15s here. Can we separate them into two code paths, and simply write:
```
#ifdef TI_LLVM_15
....
#else
....
#endif
```

Just for code cleanliness :)